starlette/responses.py

```diff
@@ -21,14 +22,21 @@
 from starlette.datastructures import URL, MutableHeaders
 from starlette.types import Receive, Scope, Send
 
+if sys.version_info >= (3, 13):  # pragma: no cover
+    from typing import TypeVar
+else:
+    from typing_extensions import TypeVar
 
-class Response:
+Content = TypeVar("Content", default=typing.Any)
+
+
+class Response(typing.Generic[Content]):
```

Comment:
I'm not changing this in this PR. There was a previous attempt to change the type to `str | bytes | None`, and it was reverted - I don't remember the reason.

Reply:
Ok. It would be good to find that reason.

Reply:
https://github.com/encode/starlette/issues/2242

Reply:
The JSONResponse should be of type Any, because the user may have customized the serializer to handle other types of data.

Also, I don't think that using "render" as a method for BaseResponse is a good idea. Methods like StreamResponse and FileResponse will not use this method at all. In Baize, I have designed corresponding classes for inheritance, which may bring some inspiration to everyone. https://github.com/abersheeran/baize/blob/master/baize/asgi/responses.py

Reply:
Thank you for sharing @abersheeran !

Reply:
> Also, I don't think that using "render" as a method for BaseResponse is a good idea.

What do you propose I do here @abersheeran ?

Reply:
> > Also, I don't think that using "render" as a method for BaseResponse is a good idea.
> 
> What do you propose I do here @abersheeran ?

I think we should derive a SmallContentResponse from BaseResponse and include the `render` method in this class. Similar to HTMLResponse, JSONResponse inherits it, while other Responses that do not require render directly inherit BaseResponse.

Reply:
But the `Response` class does have the `.render()` method... So I don't see how I'd that? I can't make the classes stop inheriting `Response`, because that would be a breaking change.

Reply:
Yes, this is also what I think is unreasonable about Starlette. For example, StreamResponse and FileReponse do not use render, but they inherit this method.

Reply:
Are you suggesting adding a breaking change then?

Reply:
Yes. 😊

Reply:
Can we do some naming to make it a non-breaking change or minimize the breakage? We could leave the render() method on the base class but mark it as deprecated vi a warning. Adding a BaseResponse class is not a breaking change. I think removing the render() method from eg StreamingResponse (because it inherits from BaseResponse and not Response) would also be okay.